File: maps/src/main/java/it/localhostsoftware/maps/huawei/model/HuaweiPointOfInterest.java
package it.localhostsoftware.maps.huawei.model;

import it.localhostsoftware.maps.model.LatLng;
import it.localhostsoftware.maps.model.PointOfInterest;

public class HuaweiPointOfInterest extends PointOfInterest<com.huawei.hms.maps.model.PointOfInterest> {
    public HuaweiPointOfInterest(com.huawei.hms.maps.model.PointOfInterest POI) {
        super(POI);
    }

    @Override
    public LatLng<?> getLatLng() {
        return new HuaweiLatLng(getPointOfInterest().latLng);
    }

    @Override
    public String getPlaceId() {
        return getPointOfInterest().placeId;
    }

    @Override
    public String getName() {
        return getPointOfInterest().name;
    }
}

File: maps/src/main/java/it/localhostsoftware/maps/huawei/model/HuaweiPointOfInterest.kt
package it.localhostsoftware.maps.huawei.model

import com.huawei.hms.maps.model.PointOfInterest

class HuaweiPointOfInterest(POI: PointOfInterest) : it.localhostsoftware.maps.model.PointOfInterest<PointOfInterest, HuaweiLatLng>(POI) {
    override val latLng: HuaweiLatLng
        get() = HuaweiLatLng(poi.latLng)
    override val placeId: String
        get() = poi.placeId
    override val name: String
        get() = poi.name
}